lib/rubyai/client.rb

```diff
@@ -1,55 +1,41 @@
-require 'faraday'
-require 'json'
-
+require 'faraday' 
+require 'json' 
+require_relative 'configuration' 
+require_relative 'http' 
+ 
 module RubyAI
   class Client
-    BASE_URL = "https://api.openai.com/v1/chat/completions"
-
-    MODELS = {
-      "gpt-4" => "gpt-4",
-      "gpt-4-0314" => "gpt-4-0314",
-      "gpt-4-32k" => "gpt-4-32k",
-      "gpt-3.5-turbo" => "gpt-3.5-turbo",
-      "gpt-3.5-turbo-0301" => "gpt-3.5-turbo-0301",
-      "text-davinci-003" => "text-davinci-003"
-    }
-
-    DEFAULT_MODEL = "gpt-3.5-turbo"
+    attr_reader :configuration
 
-    def initialize(api_key, messages, temperature: 0.7, model: DEFAULT_MODEL)
-      @api_key = api_key
-      @messages = messages
-      @temperature = temperature
-      @model = model
+    def initialize(config_hash = {})
+      @configuration = Configuration.new(config_hash)
     end
 
-    attr_accessor :api_key, :model, :messages, :temperature
-
     def call
       response = connection.post do |req|
-        req.url BASE_URL
-        req.headers['Content-Type'] = 'application/json'
-        req.headers['Authorization'] = "Bearer #{@api_key}"
-        req.body = body.to_json
+        req.url RubyAI::Configuration::BASE_URL
+        req.headers.merge!(RubyAI::HTTP.build_headers(@configuration.api_key))
```

Comment:
if we have **attr_reader :configuration** we can use everywhere just **configuration**